### ISSUE TITLE
keys and clicks are being held down during window switching

### ISSUE THREAD

**Issue by BeLikeLeBron** [bayaraa/d2gl#108]
Keys and clicks are being held down during window switching. So you can be casting something or walking somewhere and swap windows or alt+tab/win+# away while holding the click and it will continue to walk/cast in the background. Can be very dangerous and get you killed in HC. 
Reproducible
The below code fixes it for unlocked cursor users only but it's still broken for locked cursor users. i noticed it 'locks' cursor on the right side of d2 window in lobby as well though even with unlocked cursor.(credit Ajtos)

<img width="917" alt="image" src="https://github.com/bayaraa/d2gl/assets/5344347/727c094d-b484-42a4-a631-3a19386e8ec4">

tried the below for locked cursor users and it fixes it but breaks a ton of border issues with locked cursors, so it's a no-go

<img width="791" alt="image" src="https://github.com/bayaraa/d2gl/assets/5344347/40486efe-6ab4-4bea-a656-45b5eb9adbf2">

Hope that's helpful. 
Thank you! D2GL is amazing!

**Comment by youbetterdont** [comment]
I looked at this issue for awhile today and then talked with Mir, and he had a suggestion that seems to fix the issue as well as the issue with alt+tab window switching occasionally requiring an extra click.

The modifications are to the `WndProc` function in win32.cpp as follows:

```c
		case WM_ACTIVATEAPP: {
			const bool fps_capped = !App.vsync && App.foreground_fps.active;

			if (wParam) {
				App.context->setFpsLimit(fps_capped, App.foreground_fps.range.value);
				DefSubclassProc(hWnd, WM_SYSKEYUP, VK_MENU, 0);
				setCursorLock();
			} else {
				App.context->setFpsLimit(App.background_fps.active || fps_capped, App.background_fps.active ? App.background_fps.range.value : App.foreground_fps.range.value);
				if (App.window.fullscreen && App.window.auto_minimize)
					PostMessage(hWnd, WM_SYSCOMMAND, SC_MINIMIZE, 0);
				POINTS points;
				points.x = *d2::mouse_x;
				points.y = *d2::mouse_y;
				LPARAM xy = MAKELPARAM(points.x, points.y);
				SendMessage(hWnd, WM_LBUTTONUP, 0, xy);
				setCursorUnlock();
			}
			return 0;
		}
```

**Comment by ajtos** [comment]
Based on discussion with @youbetterdont , more cleaner and better long term solution would be to pass **WM_ACTIVATEAPP** message unconditionally to D2, like that:

```
		case WM_ACTIVATEAPP: {
			(...)
			break;
		}
```

Only issue is that most versions of D2, will minimize, when they receive **WM_ACTIVATEAPP** with `wParam == FALSE`, when in fullscreen. This isn't a problem for D2DX, since I assume it patches this behavior out in D2 somewhere. D2GL seems to be using most of code edits present in D2DX at first glance and I wasn't able to trace where difference is.

@IAmTrial pointed out that he has code edits that disable "auto minimize" in form of https://github.com/mir-diablo-ii-tools/SlashGaming-Diablo-II-Free-DeMin
I didn't do much testing, but with those applied to base D2 v1.10, **WM_ACTIVATEAPP** can be passed directly do underlying D2, which fixes issue with keys/mouse button being "stuck", while alt+tabbing in D2GL.

**Comment by bayaraa** [comment]
Fixed in 1.3.2.
